### DISCUSSION TITLE
Patching pains

### DISCUSSION THREAD

**Discussion by Woonters** [wixtoolset/issues#6804] [Questions]
(Current wix version being used 3.11.2.4516) 
Hey, I'm testing out creating patches in Wix, at the moment I am following (roughly) the "[Using purely Wix](https://wixtoolset.org/documentation/manual/v3/patching/wix_patching.htm)" patching path .

My example is a visual studio solution with a C# console app (simply outputting a string) and a Wix installer project, I create two msi's and each one has a unique version (1.0.0 and 1.1.0) I can create a diff.wixmst with torch.exe (reading into the wixmst reveals that the source files do have different versions that wix is picking up), the next command candle.exe works fine as well as light.exe creating a patch.wixmsp file that I think is all fine, (I can supply these files if needed). I finally run pyro.exe to which I get the warning Pyro1079, and then and error Pyro0227 saying that there are "no differences"

I know there are defiantly differences with the content of the exe, the version number of both the exe's and the msi's and even wix seems to pick this up with torch in the diff file, I haven't changed the product code (do I need to?) and obviously the upgrade code (I'm almost certain I shouldn't) 

My patch.wsx 
```
<?xml version="1.0" encoding="UTF-8"?>
<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
    <Patch 
        AllowRemoval="yes"
        Manufacturer="James" 
        MoreInfoURL="http://www.dynamocorp.com/"
        DisplayName="Sample Patch" 
        Description="Small Update Patch" 
        Classification="Update"
        >
 
        <Media Id="5000" Cabinet="RTM.cab">
            <PatchBaseline Id="RTM"/>
        </Media>
 
        <PatchFamily Id='SamplePatchFamily' Version='1.0.0.0' Supersede='yes' ProductCode="{DCF053CA-E84F-4FD0-BD13-FB030AD3F5E1}">
            <ComponentRef Id="Program01"/>
        </PatchFamily>
    </Patch>
 
    <Fragment>    

    </Fragment>
</Wix>
```
Thanks :) 

Edit: here is a test rep that shows the bug https://github.com/Fritzbox2000/WixPatchingExmaple-

**Comment by WillR27** [comment]
Hi, I have experienced the same issue but have not been able to figure out why either. I also followed the tutorial you posted.

**Comment by robmen** [comment]
I don't personally have a lot of cycles right now but linking to a GitHub repo with all the pieces in it to reproduce the problem will probably help someone look at it. Patching can be very finicky so small details can make big differences.

**Reply by Woonters** [reply]
Just got it uploaded to a github repo, hopefully easy for people to use and try for themselves : https://github.com/Fritzbox2000/WixPatchingExmaple-
